src/utils/featheryClient/integrationClient.ts

```diff
@@ -149,6 +156,104 @@ export default class IntegrationClient {
     );
   }
 
+  async triggerFlinksLoginId(loginId: string, updateFieldValues: any) {
+    await initFormsPromise;
+    const { userId } = initInfo();
+    const params = encodeGetParams({
+      form_key: this.formKey,
+      fuser_key: userId,
+      login_id: loginId
+    });
+
+    const url = `${API_URL}flinks/login-id/?${params}`;
+
+    let response: any = await this._fetch(url).catch((e) => {
+      console.error(
+        {
+          id: 'triggerFlinksLoginId',
+          message: 'failed to fetch flinks field values'
+        },
+        e
+      );
+      return null;
+    });
+
+    let pollInterval: NodeJS.Timeout;
+    let intervalCleared = false;
+
+    response = await new Promise((resolve, reject) => {
+      let innerResponse: any;
+
+      setTimeout(async () => {
+        if (!intervalCleared) {
+          clearInterval(pollInterval);
+          intervalCleared = true;
+          resolve(innerResponse);
+        }
+      }, this.FLINKS_TIMEOUT_MS);
+
+      pollInterval = setInterval(async () => {
+        innerResponse = await this._fetch(url).catch((e) => {
+          console.error(
+            {
+              id: 'triggerFlinksLoginId',
+              message: 'failed to fetch flinks field values'
+            },
+            e
+          );
+          return null;
+        });
+
+        if (innerResponse && (innerResponse as any).status === 200) {
+          clearInterval(pollInterval);
+          intervalCleared = true;
+          resolve(innerResponse);
+        } else {
+          reject(innerResponse);
```

Comment:
how is this successfully polling? if it returns a 202, wouldn't it immediately reject the promise when the promise should instead continue awaiting?

Reply:
@bo-dun-1 sorry, I missed the condition, it should be: 

```javascript 
else if (innerResponse && (innerResponse as any).status > 400) {
        clearInterval(pollInterval);
        reject(innerResponse);
}
```